src/components/AssetInput/AssetInput.tsx

```diff
@@ -147,44 +133,43 @@ const SelectedAsset: FC<AssetProps> = ({ asset, size, actions }) => {
     );
 
     return (
-        <div className="tw-relative tw-w-full tw-font-sans tw-text-s tw-bg-transparent">
+        <div
+            className="tw-relative tw-font-sans tw-w-full tw-text-s tw-bg-transparent tw-font-normal"
+            aria-labelledby={labelId}
+            title={asset.name}
+        >
             <button
                 {...mergeProps(buttonProps, focusProps)}
                 ref={buttonRef}
                 className={merge([
-                    "tw-w-full tw-flex tw-flex-row tw-flex-wrap tw-border tw-rounded tw-overflow-hidden hover:tw-border-black-90 dark:hover:tw-border-black-40 tw-group focus-visible:tw-outline-none",
-                    size === AssetInputSize.Large ? "tw-h-[11.5rem]" : "tw-h-14",
+                    "tw-w-full tw-flex tw-border tw-rounded hover:tw-border-black-90 dark:hover:tw-border-black-40 focus-visible:tw-outline-none",
+                    isFocusVisible && FOCUS_STYLE,
+                    size === AssetInputSize.Large ? "tw-h-[11.5rem] tw-flex-col" : "tw-h-14",
                     isOpen || isFocusVisible
                         ? "tw-border-black-90 dark:tw-border-black-10"
                         : "tw-border-black-20 dark:tw-border-black-80",
-                    isFocusVisible && FOCUS_STYLE,
                 ])}
             >
                 <AssetThumbnail asset={asset} size={size} isActive={isOpen || isFocusVisible} />
-                <span
+                <div
                     className={merge([
-                        "tw-flex tw-flex-1",
-                        size === AssetInputSize.Large
-                            ? "tw-h-14 tw-w-full tw-border-t tw-border-black-100 tw-border-opacity-10"
-                            : "tw-h-full",
+                        "tw-min-w-0 tw-flex tw-flex-auto tw-border-black-100 tw-border-opacity-25",
```

Comment:
```suggestion
                        "tw-min-w-0 tw-flex tw-flex-auto tw-border-black-100 tw-border-opacity-25 tw-max-w-full tw-self-stretch",
```

Needs this max-w-full otherwise it breaks on vertical stacking mode. Needs self-stretch for border to stretch the whole width in vertical too

Reply:
What do you mean by vertical stacking mode?

Reply:
AH, I mean when the size is switched to the large